include/SFML/System/Angle.hpp

```diff
@@ -0,0 +1,471 @@
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+#ifndef SFML_ANGLE_HPP
+#define SFML_ANGLE_HPP
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/System/Export.hpp>
+#include <cassert>
+
+
+namespace sf
+{
+////////////////////////////////////////////////////////////
+/// \brief Represents an angle value
+///
+////////////////////////////////////////////////////////////
+class Angle
+{
+public:
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Default constructor
+    ///
+    /// Sets the angle value to zero.
+    ///
+    ////////////////////////////////////////////////////////////
+    constexpr Angle();
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Return the angle's value in degrees
+    ///
+    /// \return Angle in degrees
+    ///
+    /// \see asRadians
+    ///
+    ////////////////////////////////////////////////////////////
+    [[nodiscard]] constexpr float asDegrees() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Return the angle's value in radians
+    ///
+    /// \return Angle in radians
+    ///
+    /// \see asDegrees
+    ///
+    ////////////////////////////////////////////////////////////
+    [[nodiscard]] constexpr float asRadians() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Return the angle wrapped to the range [-180°, 180°)
+    ///
+    /// \return Angle wrapped to [-180°, 180°)
+    ///
+    /// \see wrappedTo360
+    ///
+    ////////////////////////////////////////////////////////////
+    [[nodiscard]] constexpr Angle wrappedTo180() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Return the angle wrapped to the range [0°, 360°)
+    ///
+    /// \return Angle wrapped to [0°, 360°)
+    ///
+    /// \see wrappedTo180
+    ///
+    ////////////////////////////////////////////////////////////
+    [[nodiscard]] constexpr Angle wrappedTo360() const;
+
+    ////////////////////////////////////////////////////////////
+    // Static member data
+    ////////////////////////////////////////////////////////////
+    SFML_SYSTEM_API static const Angle Zero; //!< Predefined 0 degree angle value
```

Comment:
```suggestion
    ////////////////////////////////////////////////////////////
    // Static member data
    ////////////////////////////////////////////////////////////
    SFML_SYSTEM_API static const Angle Zero; //!< Predefined 0 degree angle value
```

I don't really see the point of this. It has the usual issue with `constexpr` support, and is it really any better than `sf::degrees(0)` or `sf::Angle{}`?

Reply:
That's fair. This follows `sf::Time`'s established pattern of a named constant for zero. I think it's nice to have but it's not that important.

Reply:
I agree with @ChrisThrasher here, and there might actually be more useful constants in the future, such as `Pi` and `Tau` (see comment [here](https://github.com/SFML/SFML/pull/1969#discussion_r785478790)).

The `constexpr` problem is something we need to find a solution for, or at least workaround -- but I don't think it should influence whether we want to have constants in our API or not. Otherwise we'd need to remove `sf::Time::Zero`, `sf::Color::*` and many others.